code/ai/aiturret.cpp

```diff
@@ -2241,8 +2272,13 @@ void ai_turret_execute_behavior(const ship *shipp, ship_subsys *ss)
 		ship_get_global_turret_info(objp, tp, &global_gun_pos, &global_gun_vec);
 	}
 
-	// Update predicted enemy position.  This used to be done in aifft_rotate_turret
-	aifft_update_predicted_enemy_pos(objp, shipp, ss, &global_gun_pos, &global_gun_vec, lep, &predicted_enemy_pos);
+	if (!in_lab) {
+		// Update predicted enemy position.  This used to be done in aifft_rotate_turret
+		aifft_update_predicted_enemy_pos(objp, shipp, ss, &global_gun_pos, &global_gun_vec, lep, &predicted_enemy_pos);
+	} else {
+		// Lab hijacks this value to use as the target position
+		predicted_enemy_pos = ss->last_aim_enemy_pos;
```

Comment:
I'd put this comment, or an additional one, on the declaration of this field as well, as I was a little confused the first time I saw you using it before I saw the explanation here.

Reply:
Ah yeah good call. I was trying to avoid adding another member just for the lab but I can do that too if that'd be clearer?

Reply:
Heh, I have the same instinct to re-use as much as possible too. It's fine as it is, in general coders in this area are likely to be much more concerned with the in-game logic code path, so minimal impact is good, just a note for others how the flow works in the lab case.

Reply:
Comment added.